scripts/update_capacity_configuration.py

```diff
@@ -0,0 +1,220 @@
+
+
+def check_capacity_config_type(capacity_config: list, config_type: type) -> None:
+    """Check that the capacity config is of the specified type"""
+    return (
+        True
+        if all(isinstance(item, config_type) for item in capacity_config)
+        else False
+    )
+
+
+def get_aggregated_capacity_config(
+    parent_zone: ZoneKey,
+) -> dict[str, Any] | None:
+    """Generate the aggregated capacity config for a parent zone"""
+    zone_keys = [zone for zone in ZONE_PARENT if ZONE_PARENT[zone] == parent_zone]
+    parent_capacity_config = ZONES_CONFIG[parent_zone]["capacity"]
+
+    all_capacity_configs = [ZONES_CONFIG[zone]["capacity"] for zone in zone_keys]
+
+    for mode in CAPACITY_MODES:
+        mode_capacity_configs = [
+            capacity_config[mode] for capacity_config in all_capacity_configs
+        ]
+
+        if not check_capacity_config_type(
+            mode_capacity_configs, type(mode_capacity_configs[0])
+        ):
+            logger.warning(
+                f"{parent_zone} capacity could not be updated because all capacity configs must have the same type"
+            )
+            return None
+        if check_capacity_config_type(mode_capacity_configs, dict):
+            parent_capacity_config[mode] = get_aggregated_capacity_config_dict(
+                mode_capacity_configs, parent_zone
+            )
+        elif check_capacity_config_type(mode_capacity_configs, list):
+            parent_capacity_config[mode] = get_aggregated_capacity_config_list(
+                mode_capacity_configs, parent_zone
+            )
+    return parent_capacity_config
+
+
+def get_aggregated_capacity_config_dict(
+    capacity_config: list[dict[str, Any]], parent_zone: ZoneKey
+) -> dict[str, Any] | None:
+    """Generate aggregated capacity config for a parent zone if the capacity config of the subzones are a dict"""
+    datetime_values = set(
+        [capacity_config["datetime"] for capacity_config in capacity_config]
+    )
+    sources = set([capacity_config["source"] for capacity_config in capacity_config])
+    if len(datetime_values) != 1:
+        logger.warning(
+            f"{parent_zone} capacity could not be updated because all capacity configs must have the same datetime values"
+        )
+        return None
+    else:
+        updated_capacity = {}
+        aggregated_value = compute_aggregated_value(capacity_config)
+        updated_capacity["datetime"] = list(datetime_values)[0]
+        updated_capacity["value"] = aggregated_value
+        updated_capacity["source"] = ", ".join(
+            [elem for elem in sources if elem is not None]
+        )
+        return sort_config_keys(updated_capacity)
+
+
+def compute_aggregated_value(capacity_config: list[dict[str, Any]]) -> float | None:
+    """Compute the aggregated capacity value as the sum of all subzone capacities"""
+    if all(capacity_config["value"] is None for capacity_config in capacity_config):
+        aggregated_value = None
```

Comment:
```suggestion
        return None
```
This way you can reduce the imbrications by one level.